SAP/spartacus
# fix: enable all existing Feature Toggles in new apps via installation schematics

The installation schematics now generates a list of available FeatureToggles in a customer's app.
It's to help customers to see what feature toggles are available in Spartacus.

Moreover, each feature toggle is explicitly enabled in customer's app, to help customers start with the most up-to-date configuration (without a technical debt of any feature toggles disabled). Of course, they can disable any feature toggle they don't want to use.

In particular, our installation schematics generates the list of FeatureToggles right in the `SpartacusFeaturesModule` module, in the `providers` array, in the client's app - all in one place.

For new apps, it's not a breaking change because they're creating a new app, not upgrading an existing one.

Note: we're not enabling any new feature toggles for existing apps. This PR is only about the installation schematics for new apps.

**QA steps**
1. build libs and publish to verdaccio
2. create new app `npx @angular/cli@17 new test-ng17-FeatureToggles
 --standalone=false`
3. add Spartacus  `ng add @spartacus/schematics --baseUrl="https://40.76.109.9:9002"`
4. verify the bottom of the `spartacus-features.module.ts` in the new app contains the enablement of all feature toggles properties (compare it with the the object `defaultFeatureToggles` from Spartacus repo):
```typescript
providers: [
  provideFeatureToggles:({
     <JSON object with all feature toggles here; each with value true>
  })
]
```

fixes https://jira.tools.sap/browse/CXSPA-6783
